client/js/entities/entityManager.js

```diff
@@ -29,16 +32,48 @@ function EntityManager(scene, conn, world, clock) {
 		}
 	};
 
-	conn.on('entity-create', function (payload) {
+	function payloadToHistoryState(payload) {
+		var kind = payload.Kind;
+		var result = { time: null, data: null };
+		if (kind === EntityKindPlayer) {
+			var bioticState = Protocol.parseBioticState(payload.State);
+			result.time = bioticState.entityState.lastUpdated;
+			result.data = bioticState;
+		} else if (kind === EntityKindWorldItem) {
+			var worldItemState = Protocol.parseWorldItemState(payload.State);
+			result.time = worldItemState.entityState.lastUpdated;
+			result.data = worldItemState;
+		} else {
+			return null;
+		}
+		return result;
+	}
```

Comment:
Double checking this...
My answer is not, definitely not. protocol is going to be responsible for transforming the binary representation of the objects into JavaScript objects. This function is taking those JavaScript objects and mapping them into another JS object specific to this file.